src/server/controllers/post.controller.ts

```diff
@@ -342,6 +350,11 @@ export const getPostHandler = async ({ input, ctx }: { input: GetByIdInput; ctx:
     const post = await getPostDetail({ ...input, user: ctx.user });
     if (!post) throw throwNotFoundError();
 
+    if (ctx.user && !ctx.user.isModerator) {
+      const blocked = await amIBlockedByUser({ userId: ctx.user.id, targetUserId: post.user.id });
+      if (blocked) throw throwNotFoundError();
+    }
+
```

Comment:
Might also wanna check the posts' intended `modelVersionId` to confirm this user can post there.

Reply:
Even if they manage to create the post, the blocking user won't see it anyways since we are filtering the content. And no notification should be created since we check on those as well